recipe-server/normandy/base/auth_backends.py

```diff
@@ -10,13 +10,19 @@
 logger = logging.getLogger(__name__)
 
 
-class LoggingModelBackend(ModelBackend):
+class LoggingAuthBackendMixin(object):
     """
-    Model-backed authentication backend that logs the results of login
-    attempts.
+    Authentication backend mixin that logs the results of login attempts.
     """
-    def authenticate(self, username=None, **kwargs):
-        result = super().authenticate(username=username, **kwargs)
+    def authenticate(self, **kwargs):
+        result = super().authenticate(**kwargs)
+
+        username = None
```

Comment:
A method that subclasses override (like `get_username` or something) would be a cleaner solution than this if statement.